src/shared/containers/challenge-detail/index.jsx

```diff
@@ -56,6 +57,10 @@ class ChallengeDetailPageContainer extends React.Component {
     const { challenge, loadChallengeDetails, loadTerms,
       openTermsModal, authTokens, challengeId } = this.props;
 
+    if (isNaN(challengeId)) {
```

Comment:
Please, replace all occurances of `isNaN(..)` by `_.isNumber(..)` from `lodash`; because https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/isNaN (especially, check `Confusing special-case behavior` section there)

Reply:
the result of `_.isNumber(NaN)` is `true`, I will use `_.isNaN` instead